packages/compass-aggregations/src/modules/input-documents.js

```diff
@@ -110,21 +122,19 @@ export const refreshInputDocuments = () => {
 
     const exampleDocumentsPipeline = [{ $limit: state.settings.sampleSize }];
 
-    if (dataService && dataService.isConnected()) {
-      dispatch(loadingInputDocuments());
-      dataService.estimatedCount(ns, options, async (error, count) => {
-        try {
-          const docs = await dataService.aggregate(ns, exampleDocumentsPipeline, options);
-          dispatch(updateInputDocuments(error ? null : count, docs, null));
-        } catch(e) {
-          dispatch(updateInputDocuments(error ? null : count, [], e));
-        }
-      });
-    } else if (dataService && !dataService.isConnected()) {
-      debug(
-        'warning: trying to refresh aggregation but dataService is disconnected',
-        dataService
-      );
+    dispatch(loadingInputDocuments());
+
+    try {
+      const data = await Promise.allSettled([
+        dataService.estimatedCount(ns, options),
+        dataService.aggregate(ns, exampleDocumentsPipeline, options)
+      ]);
+
+      const count = data[0].status === 'fulfilled' ? data[0].value : null;
+      const docs = data[1].status === 'fulfilled' ? data[1].value : [];
+      dispatch(updateInputDocuments(count, docs, null));
```

Comment:
added in [fbdf6a6](https://github.com/mongodb-js/compass/pull/3866/commits/fbdf6a6e1c5eb92a8165df8ffa8f2e59803cab8a). but i added either of the two errors instead of just aggregate.